### PR TITLE
Adds sphinx

### PR DESCRIPTION
**Pull request by Coxswain-Navigator**
## About The Pull Request
Adds sphinx, a new abnormality

- Players must solve a cipher to achieve good work rates and keep qliphoth counter high
- Petrification gaze on breach
- Players who manage to solve the puzzle are awarded with items
- These items can reverse damage caused by the abnormality or be consumed for small buffs

## Why It's Good For The Game
Is it?

## Changelog
:cl:
add: Added sphinx, a new abnormality
/:cl:

### CODE REVIEW

**Comment by Coxswain-Navigator** [comment]
To do list : visual messages when items are used, second attack for the breach, sounds, sprites, and whatever else the maintainers find issue with.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:33]: Levels are entirely unused? Level 1 agent will have nearly the same chances as level 5, ignoring temperance buff.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:53]: A list with just one line, eh?

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:171]: Why is it a loop? Just use chosenorgan.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (EgorDinamit) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:255]: Any reason it isn't a temporary visual?

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
- Rearrange procs. They should not be interrupted by other objects. Mainly speaking about line 229-237 and 251-272.
- The sprites could certainly use an update before merging.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:445]: THIS needs to be relocated, it's in the middle of a bunch of objects and very far away from Sphinx's working code.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:117]: Couldn't you just set the demand to a type and check istype like, ONCE here?

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:105]: Demand could be a pre-set type list / typecache and just pick from that.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:65]: I'm gonna say this every time, return `..()` instead of `TRUE` when doing this because returning `TRUE` makes it ignore any other checks AttemptWork runs to see if the person CAN do the work and while this may be caught elsewhere, it's still very bad practice.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:151]:
```suggestion
	var/turf/dispense_turf = get_step(src, pick(GLOB.alldirs))
```

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:197]:
```suggestion
	if(!ishuman(target))
		return
	var/mob/living/carbon/human/H = target
	if(!H.sanity_lost)
		return
	QDEL_NULL(H.ai_controller)
	H.ai_controller = /datum/ai_controller/insane/wander/sphinx
	H.InitializeAIController()
	LoseTarget(H)
```
Early returns are the standard.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:231]: There's no delay between this and the next icon_state change and the next can_act change, I'm worried this could result in it moving when it shouldn't or otherwise failing to change icon properly

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:261]: Awesome

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:26]: Duplicate Var, also _ZERO_ damage deviation?

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:226]: What this variable accomplishes could be accomplished by making it a local var to the proc, no?
Something akin to `StoneVision(attack_chain = FALSE)` and when you call it you can simply call it again with `StoneVision(TRUE)`

**Review by dsmith328 — CHANGES_REQUESTED** [review]

**Comment by Coxswain-Navigator** [comment]
Still need a sprite facelift.

File: code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm
Comment (dsmith328) [inline code/modules/mob/living/simple_animal/abnormality/waw/sphinx.dm:92]:
```suggestion
	else
		var/obj/item/I = user.get_active_held_item()
		if(!I)
			I = user.get_inactive_held_item()
		QuestHandler(I,user) //quest item must be either the active hand, or the other hand if active is empty. No guessing.
	return FALSE
```

**Review by dsmith328 — CHANGES_REQUESTED** [review]

**Review by dsmith328 — APPROVED** [review]

**Pull request merged**
